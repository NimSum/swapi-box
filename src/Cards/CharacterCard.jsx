import React from 'react';

export const CharacterCard = cards => 
  cards.map(character => (
    <article className="character-card">
      <h3>{ character.name }</h3>
      <h4>{ character.species }</h4>
      <div>
        <h4>{ character.homeworld }</h4>
        <p>{ character.population }</p>
      </div>
    </article>
  ))
