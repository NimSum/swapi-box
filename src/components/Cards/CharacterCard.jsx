import React from 'react';
import PropTypes from 'prop-types';
import { FavoriteCardBtn } from '../Buttons/Buttons';

export const CharacterCard = ({ card, updateFavoriteCount }) => 
  (
    <article key={ card.id } className="character-card">
      < FavoriteCardBtn 
       card={ card }
       updateFavoriteCount={updateFavoriteCount} />
      <h3>{ card.name }</h3>
      <h4>{ card.speciesName }</h4>
      <div className="home-planet">
        <h4>{ card.worldName }</h4>
        <p>{ card.worldPopulation }</p>
      </div>
    </article>
  )

CharacterCard.propTypes = {
  cards: PropTypes.array
}