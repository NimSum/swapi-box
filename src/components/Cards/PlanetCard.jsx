import React from 'react';

export const PlanetCard = cards => 
  <div>HIIII</div>
  // cards.map(planet => (
  //   <article className="planet-card">
  //     <h3>{ planet.name }</h3>
  //     <div className="residents">
        
  //     </div>
  //   </article>
  // ))