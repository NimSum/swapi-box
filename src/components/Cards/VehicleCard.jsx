import React from 'react';
import { FavoriteCardBtn } from '../Buttons/Buttons';
import PropTypes from 'prop-types';

export const VehicleCard = ({ card, updateFavoriteCount }) => 
  (
    <article className="vehicle-card">
      < FavoriteCardBtn 
        card={ card }
        updateFavoriteCount={updateFavoriteCount} />
      <h3>{ card.name }</h3>
      <div className="vehicle-specs">
        <p>{ card.model }</p>
        <p>{ card.vehicle_class }</p>
        <p>{ parseInt(card.passengers) + parseInt(card.crew) }</p>
      </div>
    </article>
  )

VehicleCard.propTypes = {
  card: PropTypes.object,
  updateFavoriteCount: PropTypes.func
}