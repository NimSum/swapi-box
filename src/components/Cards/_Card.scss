section.card-container {
  article {
    border-radius: 20px;
    width: 100%;
    min-width: 300px;
    margin: 0 auto;
    box-shadow: $card-shadow;
    position: relative;
    background-color: $card-bg-opacity;

    h3, h4, div, button, p {
      position: relative;
      z-index: 1;
    }

    h3, h4 {
      color: $font-card-title-color;
      font-family: $title-font;
      margin: 10px;
    }

    h3 {
      border-width: 2px 0 2px 0;
      border-style: solid;
      border-color: $card-title-border-color;
      padding: 10px 0;
      font-size: 1.7rem;
    }

    p {
      font-family: $content-font;
      margin: 2px auto;
      width: 50%;
    
    }

    img.bg-img {
      position: absolute;
      bottom: 0;
      z-index: 0;
      opacity: .1;
    }
  }


  article.character-card {
    max-height: 350px;

    div.home-planet {
      border-top: 2px solid $card-title-border-color;
      width: 30%;
      white-space: nowrap;
      margin: auto;
    }

    img.bg-img {
      width: 200px;
      left: 0;
      opacity: .2;
    }
  }

  article.vehicle-card {
    min-height: 300px;
    max-height: 350px;
    img.bg-img {
      width: 100%;
      left: 0;
      opacity: .2;
    }
  }

  article.planet-card {
    min-height: 450px;
    max-height: 600px;
    overflow: scroll;

    img.bg-img {
      width: 90%;
      top: 15%;
      left: 0;
      margin: 0 5%;
      opacity: .3;
    }
  }
}