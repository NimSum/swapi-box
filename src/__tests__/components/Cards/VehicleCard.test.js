import { mount } from 'enzyme';
import React from 'react';
import { VehicleCard } from '../../../components/Cards/VehicleCard';

describe('Character card', () => {
  const mockCard = { 
    id: 123, 
    name: 'Tesla', 
    model: 'Model 3', 
    vehicle_class: 'Wheeled', 
    passengers: '1',
    crew: '1'
  }
  
  const mockupdateFavoriteCount = jest.fn();
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      < VehicleCard 
        card={ mockCard }
        updateFavoriteCount={ mockupdateFavoriteCount }/>
    )
  })

  it('Should match snapshot with card passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })

})