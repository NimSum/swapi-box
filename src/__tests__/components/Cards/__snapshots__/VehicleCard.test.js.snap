// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character card Should match snapshot with card passed in 1`] = `
<VehicleCard
  card={
    Object {
      "crew": "1",
      "id": 123,
      "model": "Model 3",
      "name": "Tesla",
      "passengers": "1",
      "vehicle_class": "Wheeled",
    }
  }
  updateFavoriteCount={[MockFunction]}
>
  <article
    className="vehicle-card"
  >
    <FavoriteCardBtn
      card={
        Object {
          "crew": "1",
          "id": 123,
          "model": "Model 3",
          "name": "Tesla",
          "passengers": "1",
          "vehicle_class": "Wheeled",
        }
      }
      updateFavoriteCount={[MockFunction]}
    >
      <button
        className="favorite-card-btn"
        name="favorite-card"
        onClick={[Function]}
      >
        Favorite Btn(IMG)
      </button>
    </FavoriteCardBtn>
    <h3>
      Tesla
    </h3>
    <div
      className="vehicle-specs"
    >
      <p>
        Model 3
      </p>
      <p>
        Wheeled
      </p>
      <p>
        2
      </p>
    </div>
  </article>
</VehicleCard>
`;
